share/spice/nobel_prize/nobel_prize.js

```diff
@@ -0,0 +1,81 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_nobel_prize = function(api_result) {
+
+        // Validate the response (customize for your Spice)
+        if (!api_result || !api_result.prizes || api_result.prizes.length == 0) {
+            return Spice.failed('nobel_prize');
+        }
+
+        function getSearchTerm() {
+            var script = $('[src*="/js/spice/nobel_prize/"]')[0],
+            source = $(script).attr("src"),
+            query = source.match(/nobel_prize\/([^\/]+)/)[1],
+            match = decodeURIComponent(query).match(/(\d{4})?-(\w+)?/),
+            year = match && match[1],
+            category = match && match[2],
+            searchTerm = ((year && 'Year: ' + year) || "") 
+                        + " " 
+                        + ((category && ' Category: ' + category.toUpperCase()) || "");
+            
+            return [searchTerm, year, category];
+        }
+
+        function getDetailTitleNames(item) {
+            var names = [];
+            for (var laureate of item.laureates) {
```

Comment:
I'm not sure about the exact syntax for using that helper via JS, but it's definitely possible. If you take a look at the implementation you should be able to figure it out. 

You don't have to use `concat` but we do need to replace the use of a `for..of` loop